src/main/kotlin/de/jodamob/kotlin/testrunner/KotlinTestRunner.kt

```diff
@@ -1,14 +1,35 @@
 package de.jodamob.kotlin.testrunner
 
 import org.junit.runners.BlockJUnit4ClassRunner
+import org.junit.runners.model.InitializationError
 
-class KotlinTestRunner(klass: Class<*>?) : BlockJUnit4ClassRunner(load(klass)) {
+class KotlinTestRunner(klass: Class<*>) : BlockJUnit4ClassRunner(load(klass)) {
 
     companion object {
-        private  val myClassLoader = NoMoreFinalsClassLoader()
-        fun load(clazz: Class<*>?): Class<*>? {
-            val loadClass = myClassLoader.loadClass(clazz!!.name)
-            return loadClass
+
+        fun load(klass: Class<*>): Class<*>? {
+            val classOpeningClassLoader = NoMoreFinalsClassLoader(getAnnotatedClasses(klass), klass)
+            return classOpeningClassLoader.loadClass(klass.name)
+        }
+
+        @Throws(InitializationError::class)
+        private fun getAnnotatedClasses(klass: Class<*>): ClassFilter {
+            val openedPackages: List<String> = klass
+                    .getAnnotation<OpenedPackages>(OpenedPackages::class.java)
+                    ?.value
+                    .orEmpty()
+                    .toList()
+            val openedClasses = klass
+                    .getAnnotation<OpenedClasses>(OpenedClasses::class.java)
+                    ?.value
+                    .orEmpty()
+                    .toList()
+
+            if (openedPackages.isNotEmpty() || openedClasses.isNotEmpty()) {
+                return ClassFilter(openedPackages, openedClasses)
+            } else {
+                throw InitializationError("class '${klass.name}' must have a either OpenedPackages or OpenedClasses annotation")
```

Comment:
still not sure I like this one, it would mean that you always have to configure this?
Would prefer having a default mode that works on current package. Similar to Robolectric putting a warning to commandline that no manifest configured  so falling back into a default mode.
What do you think?